clang/test/SemaSYCL/sycl-device-template-diag.cpp

```diff
@@ -0,0 +1,27 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -Wno-sycl-2017-compat -fsyntax-only %s -internal-isystem %S/Inputs
+
+// This test verifies that we generate deferred diagnostics when
+// such diagnostics are in a function template.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+template <typename h>
+h *malloc_shared();
```

Comment:
```suggestion
```
This is not used now. Let's remove it as well.

Reply:
Done, thanks.